src/apps/components/AppPermissionsDialog/AppPermissionsDialogState.ts

```diff
@@ -0,0 +1,94 @@
+import { getPermissionsDiff } from "@dashboard/apps/components/AppPermissionsDialog/getPermissionsDiff";
+import { PermissionEnum } from "@dashboard/graphql";
+import { useState } from "react";
+
+type State =
```

Comment:
**nitpick**: this type name could be more descriptive

Reply:
Its not exported, doesn't context of the file provide enough context? 
Personally I find it harder to read when everything is VeryLongLocalVariable in VeryLongFile.ts. Of course when stuff gets exported, I totally agree that names should contain context

Reply:
I find names like this more difficult to find & refactor. This is a nitpick - you can ignore this if you prefer shorter names